08-translations.Rmd

```diff
@@ -0,0 +1,160 @@
+# Translations
+
+This chapter covers internationalization in R, i.e., the display of messages in languages other than English. All
+output in R (such as messages emitted by `stop()`, `warning()`, or `message()`) is eligible for translation, as
+are menu labels in the GUI. Depending on the version of R that you are using, some of the languages might already
+be available while others may need work. R leverages the [`gettext`](https://www.gnu.org/software/gettext/) program
+to handle the conversion from English to arbitrary target languages.
+
+Having messages available in other languages can be an important bridge for R learners not confident in English --
+rather than learning two things at once (coding in R and processing diagnostic information in English), they can
+focus on coding while getting more natural errors/warnings in their native tongue.
+
+The [`gettext` manual](https://www.gnu.org/software/gettext/manual/index.html) is a more canonical reference for a
+deep understanding of how `gettext` works. This chapter will just give a broad overview, with particular focus on
+how things work for R, with the goal of making it as low-friction as possible for developers and users to contribute
+new/updated translations.
+
+## How translations work
+
+Each of the default packages distributed with R (i.e., those found in `./src/library` such as `base`, `utils`,
+and `stats`) contains a `po` directory that is the central location for cataloguing/translating each package's
+messages.
+
+### .pot files
+
+The `.pot` file is a snapshot of the messages available in a given **domain**. A domain in R typically identifies
+a source package and a source language (either R or C/C++). For example, the file `R-stats.pot`
+(found in the R sources in `./src/library/stats/po`) is a catalogue of all messages produced by R code in the
```

Comment:
This might be easier to follow by first stating that stats/po contains to pot files and then explaining which ones they are and what they contain / are here for.